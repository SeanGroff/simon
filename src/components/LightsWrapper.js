// @flow
import React, { Component } from 'react';
import styled from 'styled-components';
import Light from './Light';

type Props = {
  powerOnOffAction(payload: boolean): { type: string, payload: boolean },
  startGameAction(randomNumber: number): { type: string, payload: number },
  nextTurnThunk(): any,
  power: boolean,
  counter: number,
  lightSequence: ?(number[]),
  playerTurn: boolean,
};

type State = { lightOn: number };

const LightsWrapper = styled.div`position: relative;`;

const LightsRow = styled.div`margin-bottom: -4px;`;

export default class Lights extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    (this: any).clickHandler = this.clickHandler.bind(this);
  }

  state = { lightOn: 100 };

  componentWillReceiveProps({ playerTurn, lightSequence }: Props) {
    if (lightSequence && lightSequence.length) {
      lightSequence.forEach((light: number): void => {
        console.log(light);
        this.setState((prevState, props) => ({
          lightOn: light,
        }));
      });
    }

    if (playerTurn && lightSequence && lightSequence.length) {
      lightSequence.forEach((light: number): void => {
        this.setState((prevState, props) => ({
          lightOn: 100,
        }));
      });
    }
  }

  clickHandler(id: number) {
    this.setState((prevState, props) => ({
      lightOn: id,
    }));
  }

  render() {
    return (
      <LightsWrapper>
        <LightsRow>
          <Light
            id={0}
            animate={this.state.lightOn === 0}
            topLeft
            clickable={this.props.playerTurn}
            color="#00A74A"
            lightUpColor="#13ff7c"
            onClick={() => this.clickHandler(0)}
          />
          <Light
            id={1}
            animate={this.state.lightOn === 1}
            topRight
            clickable={this.props.playerTurn}
            color="#9F0F17"
            lightUpColor="#ff4c4c"
            onClick={() => this.clickHandler(1)}
          />
        </LightsRow>
        <LightsRow>
          <Light
            id={2}
            animate={this.state.lightOn === 2}
            bottomLeft
            clickable={this.props.playerTurn}
            color="#CCA707"
            lightUpColor="#fed93f"
            onClick={() => this.clickHandler(2)}
          />
          <Light
            id={3}
            animate={this.state.lightOn === 3}
            bottomRight
            clickable={this.props.playerTurn}
            color="#094A8F"
            lightUpColor="#1c8cff"
            onClick={() => this.clickHandler(3)}
          />
        </LightsRow>
      </LightsWrapper>
    );
  }
}
