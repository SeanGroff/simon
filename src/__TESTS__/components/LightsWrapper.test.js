import React from 'react';
import { shallow } from 'enzyme';
import LightsWrapper from '../../components/LightsWrapper';

it('renders correctly', () => {
  const component = shallow(<LightsWrapper />);
  expect(component).toMatchSnapshot();
});
