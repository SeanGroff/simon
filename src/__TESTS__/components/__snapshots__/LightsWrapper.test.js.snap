// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <styled.div>
        <Light
            animate={false}
            clickable={false}
            color="#00A74A"
            id={0}
            lightUpColor="#13ff7c"
            onClick={[Function]}
            topLeft={true}
        />
        <Light
            animate={false}
            clickable={false}
            color="#9F0F17"
            id={1}
            lightUpColor="#ff4c4c"
            onClick={[Function]}
            topRight={true}
        />
    </styled.div>
    <styled.div>
        <Light
            animate={false}
            bottomLeft={true}
            clickable={false}
            color="#CCA707"
            id={2}
            lightUpColor="#fed93f"
            onClick={[Function]}
        />
        <Light
            animate={false}
            bottomRight={true}
            clickable={false}
            color="#094A8F"
            id={3}
            lightUpColor="#1c8cff"
            onClick={[Function]}
        />
    </styled.div>
</styled.div>,
  "nodes": Array [
    <styled.div>
      <styled.div>
            <Light
                  animate={false}
                  clickable={false}
                  color="#00A74A"
                  id={0}
                  lightUpColor="#13ff7c"
                  onClick={[Function]}
                  topLeft={true}
            />
            <Light
                  animate={false}
                  clickable={false}
                  color="#9F0F17"
                  id={1}
                  lightUpColor="#ff4c4c"
                  onClick={[Function]}
                  topRight={true}
            />
      </styled.div>
      <styled.div>
            <Light
                  animate={false}
                  bottomLeft={true}
                  clickable={false}
                  color="#CCA707"
                  id={2}
                  lightUpColor="#fed93f"
                  onClick={[Function]}
            />
            <Light
                  animate={false}
                  bottomRight={true}
                  clickable={false}
                  color="#094A8F"
                  id={3}
                  lightUpColor="#1c8cff"
                  onClick={[Function]}
            />
      </styled.div>
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Lights />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Lights {
        "_reactInternalInstance": [Circular],
        "clickHandler": [Function],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "lightOn": 100,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <styled.div>
                    <Light
                              animate={false}
                              clickable={false}
                              color="#00A74A"
                              id={0}
                              lightUpColor="#13ff7c"
                              onClick={[Function]}
                              topLeft={true}
                    />
                    <Light
                              animate={false}
                              clickable={false}
                              color="#9F0F17"
                              id={1}
                              lightUpColor="#ff4c4c"
                              onClick={[Function]}
                              topRight={true}
                    />
          </styled.div>
          <styled.div>
                    <Light
                              animate={false}
                              bottomLeft={true}
                              clickable={false}
                              color="#CCA707"
                              id={2}
                              lightUpColor="#fed93f"
                              onClick={[Function]}
                    />
                    <Light
                              animate={false}
                              bottomRight={true}
                              clickable={false}
                              color="#094A8F"
                              id={3}
                              lightUpColor="#1c8cff"
                              onClick={[Function]}
                    />
          </styled.div>
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <styled.div>
                    <Light
                              animate={false}
                              clickable={false}
                              color="#00A74A"
                              id={0}
                              lightUpColor="#13ff7c"
                              onClick={[Function]}
                              topLeft={true}
                    />
                    <Light
                              animate={false}
                              clickable={false}
                              color="#9F0F17"
                              id={1}
                              lightUpColor="#ff4c4c"
                              onClick={[Function]}
                              topRight={true}
                    />
          </styled.div>
          <styled.div>
                    <Light
                              animate={false}
                              bottomLeft={true}
                              clickable={false}
                              color="#CCA707"
                              id={2}
                              lightUpColor="#fed93f"
                              onClick={[Function]}
                    />
                    <Light
                              animate={false}
                              bottomRight={true}
                              clickable={false}
                              color="#094A8F"
                              id={3}
                              lightUpColor="#1c8cff"
                              onClick={[Function]}
                    />
          </styled.div>
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Lights />,
}
`;
